### PR TITLE
Added a task to check heal info before upgrade host starts.

### PR DESCRIPTION
**Pull request by rchikatw**
Added a task to check heal info before upgrade host starts.

Bug-Id: https://bugzilla.redhat.com/show_bug.cgi?id=1721366

### CODE REVIEW

**Comment by ovirt-infra** [comment]
Hello contributor, thanks for submitting a PR for this project! 

I am the bot who triggers "standard-CI" builds for this project.
As a security measure, I will not run automated tests on PRs that are not from white-listed contributors.

In order to allow automated tests to run, please ask one of the project maintainers to review the code and then do one of the following:

1. Type `ci test please` on this PR to trigger automated tests for it.
2. Type `ci add to whitelist` on this PR to trigger automated tests for it and also add you to the contributor white-list so that your future PRs will be tested automatically. ( keep in mind this list might be overwritten if the job XML is refreshed, for permanent whitelisting, please follow #3 option )
3. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the project organization, so you'll be able to run tests for all the organization's projects.

**Comment by mnecas** [comment]
Please resolve the merge conflicts

File: defaults/main.yml [hunk 1]
```diff
@@ -11,3 +11,6 @@ host_statuses:
 host_names:
   - '*'
 pinned_vms_names: []
+heal_retries: 6
+heal_delay: 300
+gluster_heal_wait: 5
```
Comment (mnecas) [inline defaults/main.yml]: Add them to README.md

Comment (rchikatw) [reply defaults/main.yml]: Added

**Comment by rchikatw** [comment]
> Please resolve the merge conflicts

Resolved.

**Comment by mnecas** [comment]
LGTM @mwperina

File: README.md [hunk 2]
```diff
@@ -41,6 +41,9 @@ Role Variables
 | check_upgrade           | false                 | If true, run check_for_upgrade action on all hosts before executing upgrade on them. If false, run upgrade only for hosts with available upgrades and ignore all other hosts. |
 | reboot_after_upgrade    | true                  | If true reboot hosts after successful upgrade. |
 | use_maintenance_policy  | true                  | If true the cluster policy will be switched to cluster_maintenance during upgrade otherwise the policy will be unchanged. |
+| heal_retries            | 6                     | Maximum Number of times heal check to be done if it fails. |
```
Comment (mwperina) [inline README.md]: Please rephrase: `Maximum number of attempts to check if gluster healing is still in progress`. And I'd probably also rename the option to `healing_in_progress_checks`

File: README.md [hunk 3]
```diff
@@ -41,6 +41,9 @@ Role Variables
...
+| heal_delay              | 300                   | The Interval between each retry for heal to check    |
```
Comment (mwperina) [inline README.md]: Please rephrase: `The delay in seconds between each attempt to check if gluster healing is still in progress`. And I'd also rename the option to `healing_in_progress_check_delay`

File: tasks/upgrade.yml [hunk 4]
```diff
@@ -61,6 +61,17 @@
...
+  uri:
+    url: "{{ ovirt_auth.url }}/clusters/{{ cluster_info.ovirt_clusters[0].id }}/refreshglusterhealstatus"
+    method: POST
+    body_format: json
+    validate_certs: false
+    headers:
+      Authorization: "Bearer {{ ovirt_auth.token }}"
+    body: "{}"
+  when: cluster_info.ovirt_clusters[0].gluster_service | bool
+
```
Comment (mwperina) [inline tasks/upgrade.yml:76]: That's super ugly. Could you please post a patch to ovirt_cluster module to expose this action properly?

File: tasks/upgrade.yml [hunk 5]
```diff
@@ -70,6 +81,11 @@
     reboot_after_upgrade: "{{ reboot_after_upgrade }}"
     timeout: "{{ upgrade_timeout }}"
 
+- name: Gluster heal wait check when multiple hosts are present, Heal take some time to update entries after host is up
+  pause:
+    minutes: "{{ gluster_heal_wait }}"
+  when:  cluster_info.ovirt_clusters[0].gluster_service | bool and host_info.ovirt_hosts | length > 1
```
Comment (mwperina) [inline tasks/upgrade.yml]: Please reformat to make it more readable:

```
when:
  - cluster_info.ovirt_clusters[0].gluster_service|bool
  - host_info.ovirt_hosts|length > 1
```

File: README.md [hunk 6]
```diff
@@ -41,6 +41,9 @@ Role Variables
...
+| gluster_heal_wait       | 5                     | Waiting time after host upgrade finish.             |
```
Comment (mwperina) [inline README.md]: Please rename to `wait_to_finish_healing` and rephrase the description: `Delay in minutes to wait to finish gluster healing process after successful host upgrade`

**Review by mwperina — CHANGES_REQUESTED** [review]

File: tasks/upgrade.yml (hunk 4 above)
Comment (mnecas) [reply tasks/upgrade.yml:76]: This feature was added just last month [ClusterService.java](https://github.com/oVirt/ovirt-engine-api-model/blob/master/src/main/java/services/ClusterService.java#L426).
I'm not sure what to do with it. Probably revert changes of the previous PR and wait till 4.4 will become a minimal requirement. What do you think @mwperina?

Comment (rchikatw) [reply tasks/upgrade.yml:76]: @mwperina My engine code is going to merge in next few day's @Ori Liel Working on it he facing some issue for next version release. I will do the changes in ovirt_cluster module and raise PR. Whenever sdk4.4 will release, will update this particular task.. In Cluster Upgrade Role.

Comment (mwperina) [reply tasks/upgrade.yml:76]: OK, let's have it as right now and let's fix it once within ovirt-ansible-collection

**Comment by rchikatw** [comment]
@mwperina  I have updated the suggested comment. Please merge if it looks fine.

**Review by mwperina — APPROVED** [review]
+1

**Review by mnecas — APPROVED** [review]
LGTM

**Pull request merged**